Dockerfile

```diff
@@ -67,7 +67,16 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 && apt-get update && apt-get install -y vault \
 # Give Vault the ability to use the mlock syscall without running the process as root. The mlock syscall prevents memory from being swapped to disk.
 # Explanation: https://github.com/hashicorp/vault/issues/10048#issuecomment-700779263
-&& setcap cap_ipc_lock= /usr/bin/vault
+&& setcap cap_ipc_lock= /usr/bin/vault \
+# Install Docker CLI
+&& curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
+&& echo \
+  "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+  $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null \
+&& apt-get update \
+&& apt-get install -y --no-install-recommends \
+    docker-ce=5:20.10.\* \
```

Comment:
Use kaniko https://github.com/GoogleContainerTools/kaniko for unprivileged docker container builds.

Reply:
The Azure DevOps Docker@2 tasks depend on Docker directly unfortunately.
Writing a task in Azure DevOps script that uses Kaniko will take more time, without much added value.
Are you sure we want to do that?

Reply:
Eh, let me take a look.

Reply:
Apparently to run the Kaniko build job we would still need to have the Docker CLI in the ADO image.
With that, the Azure DevOps Docker task is not versatile enough to provide file inputs to containers, so we cannot use Kaniko with Azure DevOps.

The example that describes why we need files is [here](https://blog.csanchez.org/2020/09/30/building-docker-images-with-kaniko-pushing-to-azure-container-registry-acr/).